components/ui/checkbox.tsx

```diff
@@ -0,0 +1,58 @@
+'use client'
+
+import * as React from 'react'
+import * as CheckboxPrimitive from '@radix-ui/react-checkbox'
+import { Check } from 'lucide-react'
+import { cn } from '@/lib/shadCnUtil'
+import { InputProps } from './input'
+import { ModalText } from './modal'
+
+export interface CheckboxWithTextProps extends InputProps {
+  text: string
+}
+
+const Checkbox = React.forwardRef<
+  React.ElementRef<typeof CheckboxPrimitive.Root>,
+  React.ComponentPropsWithoutRef<typeof CheckboxPrimitive.Root>
+>(({ className, ...props }, ref) => (
+  <CheckboxPrimitive.Root
+    ref={ref}
+    className={cn(
+      'peer h-4 w-4 shrink-0 rounded-[2px] border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground',
+      className
+    )}
+    {...props}
+  >
+    <CheckboxPrimitive.Indicator
+      className={cn(
+        'flex items-center justify-center text-black bg-white outline-none'
+      )}
+    >
+      <Check className="h-auto w-auto" />
+    </CheckboxPrimitive.Indicator>
+  </CheckboxPrimitive.Root>
+))
+Checkbox.displayName = CheckboxPrimitive.Root.displayName
+
+const CheckboxWithText = React.forwardRef<
+  HTMLInputElement,
+  CheckboxWithTextProps
+>(({ className, text, id, type, ...props }, ref) => {
+  return (
+    <div
+      className={cn('flex flex-row align-middle gap-2', className)}
+      ref={ref}
+      {...props}
+    >
+      <div className="flex pt-[6px]">
+        <Checkbox />
+      </div>
+      <label>
+        <ModalText className="pb-1" text={text} />
+      </label>
```

Comment:
This will allow you to click the label text to check it
```suggestion
      <label className="flex pt-[6px] gap-3 items-center">
        <Checkbox />
        <ModalText text={text} />
      </label>
```